tests/warn/i22232.scala

```diff
@@ -0,0 +1,19 @@
+class C
+object C:
+  extension (c: C) def equals(that: Any): Boolean = false // warn
+
+object X:
+  class C
+  opaque type D <: C = C
+  object D:
+    extension (d: D) def equals(that: Any): Boolean = false // warn
+
+object Upperbound:
+  opaque type IArray[+T] <: String = String
```

Comment:
I wanted to suggest to change the name of the opaque type, I think we can all agree that `IArray[+T] <: String = String` is weird... Maybe:
```suggestion
  opaque type MyString <: String = String
```